src/crimson/osd/osd_operations/scrub_events.cc

```diff
@@ -258,9 +258,10 @@ ScrubScan::ifut<> ScrubScan::deep_scan_object(
   DEBUGDPP("obj: {}", pg, obj);
   using crimson::common::local_conf;
   auto &entry = ret.objects[obj.hobj];
+  auto progress_ref = std::make_unique<obj_scrub_progress_t>();
+  auto &progress = *progress_ref;
   return interruptor::repeat(
-    [FNAME, this, progress = obj_scrub_progress_t(),
-     &obj, &entry, &pg]() mutable
+    [FNAME, this, &progress, &obj, &entry, &pg]()
```

Comment:
as a follow-up, is it possible to transform the `interruptor::repeat()` block to a plain loop in C++, now that the coroutine supported is added to crimson?

Reply:
The next time I make a large enough change here, the whole thing will be a coroutine with normal loops.